website/docs/client/python.md

```diff
@@ -60,6 +63,123 @@ The following environment variables are available to use:
 
 If you are using Airflow integration, there are additional [environment variables available](../integrations/airflow/usage.md#environment-variables).
 
+#### Dynamic configuration with environment variables
+
+You can also configure the client with dynamic environment variables.
+Environment variables that configure the OpenLineage client follow a specific pattern. All variables that affect the client configuration start with the prefix `OPENLINEAGE__`, followed by nested keys separated by double underscores (`__`).
+
+##### Key Features
+
+1. Prefix Requirement: All environment variables must begin with `OPENLINEAGE__`.
+2. Sections Separation: Configuration sections are separated using double underscores `__` to form the hierarchy.
+3. Lowercase Conversion: Environment variable values are automatically converted to lowercase.
+4. JSON String Support: You can pass a JSON string at any level of the configuration hierarchy, which will be merged into the final configuration structure.
+5. Hyphen Restriction: You cannot use `-` in environment variable names. If a name strictly requires a hyphen, use a JSON string as the value of the environment variable.
```

Comment:
```suggestion
5. Hyphen Restriction: Since environment variable names cannot contain `-` (hyphen), if a name strictly requires a hyphen, use a JSON string as the value of the environment variable.
```